TechnologyEnhancedLearning/DLSV2
# HEEDLS-354 - Reset Password email sending implemented.

ForgotPassword controller has the post functionality implemented with some basic validation. This is hooked up to a new PasswordResetService that does most of the logic. A new basic UserService for getting AdminUsers or Delegates (Candidates) was introduced, as has an EmailService. 

Note that the NotificationService has been modified to now use the EmailService and has been sanity checked, but not properly tested.

Succeeded tests of:
- Manually tested IE11, Edge, Chrome, Firefox*
- Manually tested Chrome mobile view
- Manually tested No JS
- Tested with screen reader
- Ran all unit tests successfully.
*Firefox has a very old bug (see: https://bugzilla.mozilla.org/show_bug.cgi?id=308064 ) that prevents the error summary link from focusing on the input textbox as required by the NHS error summary description - discussion in the Show & Tell on 26th March 2021 confirmed that no workaround should be performed for this.

UI Change (/ForgotPassword page with error messages):
![ForgotPassword with errors](https://user-images.githubusercontent.com/80777689/112627382-7274e980-8e29-11eb-901e-904c5d14358b.PNG)
